include/seastar/http/client.hh

```diff
@@ -19,10 +19,70 @@
  * Copyright (C) 2022 Scylladb, Ltd.
  */
 
+#include <seastar/net/api.hh>
+#include <seastar/core/iostream.hh>
+
 namespace seastar {
 
 namespace http {
 
+class request;
+class reply;
+
+class client {
+    connected_socket _fd;
+    input_stream<char> _read_buf;
+    output_stream<char> _write_buf;
+
+public:
+    /**
+     * \brief Create an http client
+     *
+     * Construct the client that will work over the provided \fd transport socket
+     *
+     */
+    client(connected_socket&& fd);
+
+    /**
+     * \brief Send the request and wait for response
+     *
+     * Sends the provided request to the server, and returns a future that will resolve
+     * into the server response.
+     *
+     * If the request was configured with the set_expects_continue() and the server replied
+     * early with some error code, this early reply will be returned back.
+     *
+     * The returned reply only contains the status and headers. To get the reply body the
+     * caller should read it via the input_stream provided by the client.in() method.
+     *
+     * \param rq -- request to be sent
+     *
+     */
+    future<reply> make_request(request rq);
+
+    /**
+     * \brief Get a reference on the client input stream
+     *
+     * The stream can be used to get back the server responce body. After the stream is
+     * finished, the reply can be additionally updated with trailing headers and chunk
+     * extentions
+     *
+     */
+    input_stream<char> in(reply& rep);
```

Comment:
Method of reply?

Reply:
Can't be _that_ nice :( it takes the stream from client to wrap it into the returning stream. So even if it's the reply method it would look like `input_stream<char> reply::in(client& cln);` which doesn't look much better.

I had an idea to make client::make_request return both -- reply and the stream with body like `future<std::pair<reply, input_stream<char>> client::make_request(request);`

Which of those three do you think looks the least ugly?